### PR TITLE
Reduce unnecessary joins when retrieving discussions

### PR DESCRIPTION
**Pull request by iHiD**
@glennj This **might** help if you're getting sporadic slow requests here, but it's hard to test.

### CODE REVIEW

**Review by ErikSchierboom — APPROVED** [review]
LGTM! There was a build failure: https://github.com/exercism/website/runs/7887521652?check_suite_focus=true
This actually showed that less data is retrieved by default! I fixed it in https://github.com/exercism/website/pull/2864/commits/90b54f218f01ea50d410182d53c9056a7aa4445f

File: app/controllers/api/mentoring/discussions_controller.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def tracks
         current_user,
         params[:status],
         sorted: false, paginated: false
-      ).group(:track_id).count
+      ).joins(solution: [exercise: :track]).group(:track_id).count
```
Comment (iHiD) [inline app/controllers/api/mentoring/discussions_controller.rb]:
```suggestion
      ).joins(solution: {exercise: :track}).group(:track_id).count
```

**Comment by iHiD** [comment]
Nice - thanks for fixing! Updated the syntax slightly 🙂

File: app/controllers/api/mentoring/discussions_controller.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ def tracks
         current_user,
         params[:status],
         sorted: false, paginated: false
-      ).group(:track_id).count
+      ).joins(solution: {exercise: :track}).group(:track_id).count
```
Comment (iHiD) [inline app/controllers/api/mentoring/discussions_controller.rb]:
```suggestion
      ).joins(solution: { exercise: :track }).group(:track_id).count
```

**Pull request merged**